Updates the documentation of the ownership. (#455)
AutoPas/AutoPas: README.md

TODO

### Particle Ownership
Particles saved in an AutoPas container can be one of two possible states:

> * owned: Particles that belong to this AutoPas instance. These particles are either inside of the boundary of the AutoPas instance or very close to the boundary (less than a distance of skin/2 away). If a particle is added via `addParticle()`, it is automatically added as an owned particle. An owned particle can explicitly be removed by deleting the particle using an iterator (`autoPas.deleteParticle(iterator)`). On update of the AutoPas container (using `updateContainer()`) owned particles that move outside of the boundary of its parent, AutoPas container are returned.

Reviewer: ```suggestion
* owned: Particles that belong to this AutoPas instance. These particles are either inside of the boundary of the AutoPas instance or very close to the boundary (less than a distance of skin/2 away). If a particle is added via `addParticle()`, it is automatically added as an owned particle. An owned particle can explicitly be removed by deleting the particle using an iterator (`autoPas.deleteParticle(iterator)`). On update of the AutoPas container (using `updateContainer()`) owned particles that move outside of the boundary of its parent AutoPas container are returned.
```
(the `,` between parent and AutoPas at the end is misleading imho)

### Simulation Loop
One simulation loop should always consist of the following phases:

1. Updating the Container:
   ```C++
   auto [invalidParticles, updated] = autoPas.updateContainer();
   ```
   This call will potentially trigger an update of the container inside of AutoPas. The update will be performed if either
   
    a. Enough samples are collected for the current configuration OR
    
    b. The rebuild frequeny is reached.
   
   If the update is performed, the returned bool `updated` is `true` and the returned vector `invalidParticles` consists of the particles that are leaving this container. These are particles which were previously owned by this AutoPas container, but have left the boundary of this container, i.e., their current position resides outside of the container.
   
   If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.

>    An update is sometimes skipped to ensure that containers do not change and to enable the best possible performance for containers that use neighbor lists.

Reviewer: Justify why we want containers sometimes not to change.

>    If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.
>    An update is sometimes skipped to ensure that containers do not change and to enable the best possible performance for containers that use neighbor lists.

Author: ```suggestion
   If the update is not performed, `updated` will be false and the returned vector `invalidParticles` will be empty.
   An update is sometimes skipped to ensure that containers do not change, which allows containers to reuse neighbor lists thus enabling better performance.
```
@FG-TUM is this fine by you?

Reviewer: yes

2. Handling the leaving particles

>    * This step can be skipped if `updated` was false. If you use multiple MPI instances, you have to ensure that all instances rebuild at the same time steps. This is guaranteed, if the sampling frequency is the same as (or a multiple of) the rebuild frequency.

Reviewer: ```suggestion
   * This step can be skipped if `updated` was false. If you use multiple MPI instances, you have to ensure that all instances rebuild during the same time step. This is guaranteed if the sampling frequency is the same as (or a multiple of) the rebuild frequency.
```
